opensoundscape/preprocess/preprocessors.py

```diff
@@ -138,23 +140,34 @@ def forward(
             "_start_time": None if clip_times is None else clip_times["start_time"],
             "_sample_duration": self.sample_duration,
             "_preprocessor": self,
+            "_trace": self.pipeline.copy(deep=True) if trace else None,
```

Comment:
why do we need a copy of the entire pipeline? Or is this just a way of duplicating the indices of the pipeline dictionary?